src/test/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfigTest.java

```diff
@@ -45,4 +47,20 @@ public void testAllowlist() {
         assertTrue(config.isAllowed("kafka_server_metric"));
         assertTrue(config.isAllowed("kafka_network_metric"));
     }
+
+    @Test
+    public void testListenerParseListener() {
+        assertEquals(new PrometheusMetricsReporterConfig.Listener("", 8080), PrometheusMetricsReporterConfig.Listener.parseListener("http://:8080"));
+        assertEquals(new PrometheusMetricsReporterConfig.Listener("123", 8080), PrometheusMetricsReporterConfig.Listener.parseListener("http://123:8080"));
+        assertEquals(new PrometheusMetricsReporterConfig.Listener("::1", 8080), PrometheusMetricsReporterConfig.Listener.parseListener("http://::1:8080"));
+        assertEquals(new PrometheusMetricsReporterConfig.Listener("::1", 8080), PrometheusMetricsReporterConfig.Listener.parseListener("http://[::1]:8080"));
+        assertEquals(new PrometheusMetricsReporterConfig.Listener("random", 8080), PrometheusMetricsReporterConfig.Listener.parseListener("http://random:8080"));
```

Comment:
Probably worth adding a test like:
`assertThrows(ConfigException.class, () -> PrometheusMetricsReporterConfig.Listener.parseListener("abchttp://:8080asd"));`

This would currently fail because we're using `matcher.find()` instead of matcher.matches() in `parseListener()`

Reply:
Added

Reply:
To be exhaustive we can also try with `randomhttp://:8080` and `http://:8080random`

Reply:
Added